core/src/main/java/io/el/concurrent/DefaultEventLoopGroup.java

```diff
@@ -0,0 +1,186 @@
+            try {
+              while (!el.isTerminated()) {
+                el.awaitTermination(Integer.MAX_VALUE, TimeUnit.SECONDS);
+              }
+            } catch (InterruptedException interrupted) {
+              Thread.currentThread().interrupt();
+              break;
+            }
+          }
+        }
+        throw new IllegalStateException("failed to create a child event loop", e);
+      }
+    }
+
+    this.chooser = chooserFactory.newChooser(this.children);
+  }
+
+  protected abstract EventLoop newChild(Executor executor) throws Exception;
+
+  protected abstract ThreadFactory newDefaultThreadFactory();
+
+  @Override
+  public EventLoop next() {
+    return this.chooser.next();
+  }
+
+  @Override
+  public Iterator<EventLoop> iterator() {
+    return this.children.iterator();
+  }
+
+  @Override
+  public boolean shutdownGracefully(long timeout, TimeUnit unit) {
+    return this.children.stream()
+        .map(c -> c.shutdownGracefully(timeout, unit))
+        .filter(s -> s.equals(false))
+        .findAny()
+        .orElse(true);
+  }
+
+  @Override
+  public void shutdown() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public List<Runnable> shutdownNow() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean isShuttingDown() {
+    return this.children.stream()
+        .map(EventLoop::isShuttingDown)
+        .filter(s -> s.equals(false))
+        .findAny()
+        .orElse(true);
+  }
+
+  @Override
+  public boolean isShutdown() {
+    return this.children.stream()
+        .map(EventLoop::isShutdown)
+        .filter(s -> s.equals(false))
+        .findAny()
+        .orElse(true);
+  }
+
+  @Override
+  public boolean isTerminated() {
+    return this.children.stream()
+        .map(EventLoop::isTerminated)
+        .filter(s -> s.equals(false))
+        .findAny()
+        .orElse(true);
+  }
+
+  /***
+   * Wait until every child {@link EventLoop} terminated through while-loop
+   */
+  @Override
+  public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
+    long deadline = System.nanoTime() + unit.toNanos(timeout);
+    for (EventLoop el : this.children) {
+      long timeLeft = deadline - System.nanoTime();
+      while (timeLeft > 0 && !el.awaitTermination(timeLeft, TimeUnit.NANOSECONDS)) {
+        timeLeft = deadline - System.nanoTime();
+      }
+      if (timeLeft <= 0) {
+        break;
+      }
+    }
+    return this.isTerminated();
+  }
+
+  @Override
+  public <V> Promise<V> submit(Callable<V> task) {
+    return this.next().submit(task);
+  }
+
+  @Override
+  public <T> Future<T> submit(Runnable task, T result) {
+    return this.next().submit(task, result);
+  }
+
+  @Override
+  public Promise<?> submit(Runnable task) {
+    return this.next().submit(task);
+  }
+
+  @Override
+  public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks)
```

Comment:
All invoke series methods are from `ExecutorService` these will not be used I think. So let's move these methods to AbstractEventLoop and throw UnsupportedException